### DISCUSSION TITLE
Capabilities with PSD -- is ImageMagick suitable?

### DISCUSSION THREAD

**Discussion by lex3001** [ImageMagick/ImageMagick#3024] [Help]
Hi,

I have **PSD** files coming from another team and they have many **layers** in them, such as solid fill layers with a mask. I am looking to programmatically (e.g. from an unattended server process) import the PSD, make some **changes to some layers** (such as changing a fill color), and then **export the _new composite_ as a PNG image**. I do not need to export the PSD with the changes, although that might be useful for troubleshooting. Would ImageMagick be suitable for this task?
There are some additional tasks that are on the roadmap as well, so I am also interested in these:
1. adjust the color of a solid fill layer w/ mask (as mentioned above)
2. replace sections of a raster layer with a new image (sort of like copying and pasting a rectangle over part of an existing image layer)
3. change the text in a text layer (e.g. text along a path, etc.)

I am guessing that if ImageMagick can load the PSD with these kinds of layers, allow them to be manipulated, and output the composite image reflecting the changes, it will suit our technical needs as well. Some existing tools I have tried will simply export the composite image from the PSD, but that was generated when the PSD file was saved in PhotoShop and does reflect changes being made. Scripting PhotoShop itself is not a viable option for various reasons I won't get into. I'd also like to avoid an overhaul to the business process -- e.g. I'd like the designers to be able to keep using PhotoShop to create the source files and save them as PSD.

**Comment by fmw42** [comment]
ImageMagick is a raster processor. If it can even load your PDF, it will rasterize it and you will not be able edit any part of it, since all the vector data is lost. I suggest you get Adobe Acrobat.

**Reply by lex3001** [reply]
Hi, sorry if I wasn't clear, but I am talking about PSD files which are an Adobe PhotoShop format, not an Acrobat / PDF format. PSD is a supported format -- https://imagemagick.org/script/formats.php -- but I am not sure how far that support extends and what you can do with manipulating layers and compositing in ImageMagick.

**Reply by fmw42** [reply]
Sorry, my mistake. Imagemagic can read PSD files with simple layers. But editing them is not simple. Layers have to be separated, processed and then recombined. I do not think ImageMagick can edit text in PSD layers. You can write over the region with text and then add new text. It can change colors with a mask and can swap images. But it is not a GUI process and is a bit more complicated than editing in Photoshop.
